akka-sample-fsm-java/src/main/java/sample/Hakker.java

```diff
@@ -77,15 +76,15 @@ private Hakker(ActorContext<Command> ctx, String name, ActorRef<Chopstick.Comman
    //it starts to wait for the response of the other grab
    private Behavior<Command> hungry() {
       return Behaviors.receive(Command.class)
-         .onMessage(AdaptedAnswer.class, m -> m.msg.chopstick.equals(left), (msg) ->
+         .onMessage(HandleChopstickAnswer.class, m -> m.msg.isBusy(), (msg) ->
+            firstChopstickDenied()
+         )
```

Comment:
The actual fix, Scala would match on busy/taken but here we could accidentally move on with a chopstick that was busy before